src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/AsylumCaseStatusSearchHandler.java

```diff
@@ -35,9 +46,16 @@ public boolean canHandle(
 
         final CaseDetails<AsylumCase> caseDetails = callback.getCaseDetails();
         final AsylumCase asylumCase = caseDetails.getCaseData();
-
-        //Should POST to home office service with search parameters
-        //Should return updated AsylumCase
+        final Optional<String> homeOfficeReferenceNumber = asylumCase.read(
+            AsylumCaseDefinition.HOME_OFFICE_REFERENCE_NUMBER);
+
+        if (homeOfficeReferenceNumber.isPresent()) {
```

Comment:
We should probably think about this a bit more e.g. should we consider throwing an error here?  If there is no homeOfficeReferenceNumber then should we be just returning the same asylum case or something to tell the client that they have not provided mandatory information.

Reply:
Throw IllegalStateException by checking and then call API. Test updated

Reply:
What is the http code we are returning here to the client app?

Reply:
It returns 500. I updated EndpointSecurityTest to have homeofficeReference number. Also, throw an exception before calling API
if (!homeOfficeReferenceNumber.isPresent()) {
            throw new IllegalStateException("Home office reference number cannot be null");
        }